pkg/services/login/authinfo.go

```diff
@@ -84,29 +85,20 @@ func IsExternallySynced(cfg *setting.Cfg, authModule string) bool {
 		return false
 	}
 	switch authModule {
-	case GoogleAuthModule:
-		return !cfg.GoogleSkipOrgRoleSync
-	case OktaAuthModule:
-		return !cfg.OktaSkipOrgRoleSync
-	case AzureADAuthModule:
-		return !cfg.AzureADSkipOrgRoleSync
-	case GitLabAuthModule:
-		return !cfg.GitLabSkipOrgRoleSync
-	case GithubAuthModule:
-		return !cfg.GitHubSkipOrgRoleSync
-	case GrafanaComAuthModule:
-		return !cfg.GrafanaComSkipOrgRoleSync
-	case GenericOAuthModule:
-		return !cfg.GenericOAuthSkipOrgRoleSync
+	case GoogleAuthModule, OktaAuthModule, AzureADAuthModule, GitLabAuthModule, GithubAuthModule, GrafanaComAuthModule, GenericOAuthModule:
+		if oauthInfo == nil {
+			return false
+		}
+		return !oauthInfo.SkipOrgRoleSync
```

Comment:
we still need to decipher which provider is turned of and turned on sync.

Reply:
I am pretty sure this function covers that right. I still check `IsProviderEnabled`

Reply:
The provider is enabled. Detection of skip org role sync set in the configuration is not picked up from what I see.

We want to be able to set

auth.google
skip org role sync = true

For only Google